cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -258,7 +264,23 @@ private int runTestSuite(Target target, Package currentPackage, JBallerinaBacken
         String orgName = currentPackage.packageOrg().toString();
         String classPath = getClassPath(jBallerinaBackend, currentPackage);
         List<String> cmdArgs = new ArrayList<>();
-        cmdArgs.add(System.getProperty("java.command"));
+        boolean nativeImageAgentEnabled = false;
+        if (enableNativeImageAgent) {
+            String graalvmHome = System.getenv("GRAALVM_HOME");
+            if (graalvmHome != null) {
+                nativeImageAgentEnabled = true;
+                String javaCmd = graalvmHome + File.separator + BIN_DIR_NAME + File.separator
+                        + (OS.contains("win") ? "java.exe" : "java");
+                cmdArgs.add(javaCmd);
+                cmdArgs.add("-agentlib:native-image-agent=config-output-dir=target/native-image-configs");
```

Comment:
Users can pass a custom directory for the target. So we should always access the target from Project.targetDir()

Reply:
Fixed via [b274375](https://github.com/ballerina-platform/ballerina-lang/pull/39207/commits/b27437522da4d101d4a4e2c31b711fec2a49d5a1)